src/utilities/createRouterRoutes.ts

```diff
@@ -51,6 +55,17 @@ export function createRouterRoutes(routes: Routes, parentContext: ParentContext
   }, [])
 }
 
-function reduceParams(entries: Path[] | Query[]): Record<string, Param[]> {
-  return entries.reduce((params, entry) => mergeMaybeTuples(params, entry.params as Record<string, Param | Param[]>), {})
+function hasDuplicateParams(path: Path, query: Query): boolean {
+  const keys = [...Object.keys(path.params), ...Object.keys(query.params)]
+
+  return new Set(keys).size !== keys.length
```

Comment:
Creating a `set` is crazy slow. How many times are we expecting this utility to get called when creating routes? Once per route?

Reply:
here's an alternative that doesn't use `Set`. Do you prefer this?

```ts
  const pathKeys = Object.keys(path)
  const queryKeys = Object.keys(query)

  return pathKeys.some(key => queryKeys.includes(key))
  ```

Reply:
That would be significantly faster especially considering there are likely very few keys. I'd go with that version.